src/Microsoft.DotNet.CodeFormatting.Tests/PropertyBagTests.cs

```diff
@@ -0,0 +1,441 @@
+                    }
+                    Assert.AreEqual(expected[key], actual[key]);
+                }
+            }
+
+            private static void ValidateStringSet(StringSet expected, StringSet actual)
+            {
+                Assert.AreEqual(expected.Count, actual.Count);
+                foreach (string value in expected)
+                {
+                    Assert.IsTrue(actual.Contains(value));
+                }
+            }
+
+            internal void InitializePropertyBag(PropertyBag propertyBag)
+            {
+                propertyBag.SetProperty(DoubleOption, DoubleValue);
+                propertyBag.SetProperty(BooleanOption, BooleanValue);
+                propertyBag.SetProperty(TestEnumOptionThree, TestEnumValue);
+                propertyBag.SetProperty(StringSetOption, StringSetValue);
+
+                PropertyBagValue = new PropertyBag();
+                PropertyBagValue.SetProperty(DoubleOption, EmbeddedDoubleValue);
+                PropertyBagValue.SetProperty(BooleanOption, EmbeddedBooleanValue);
+                PropertyBagValue.SetProperty(TestEnumOptionThree, EmbeddedTestEnumValue);
+                PropertyBagValue.SetProperty(StringSetOption, EmbeddedStringSetValue);
+
+
+                propertyBag.SetProperty(PropertyBagOption, PropertyBagValue);
+
+                TypedPropertyBagValue = propertyBag.GetProperty(TypedPropertyBagOption);
+                TypedPropertyBagValue.SetProperty(TestEnumOptionThree, EmbeddedTestEnumValue);
+
+                propertyBag.SetProperty(TypedPropertyBagOption, TypedPropertyBagValue);
+            }
+
+            internal void ValidatePropertyBag(PropertyBag propertyBag)
+            {
+                Assert.AreEqual(DoubleValue, propertyBag.GetProperty(DoubleOption));
+                Assert.AreEqual(BooleanValue, propertyBag.GetProperty(BooleanOption));
+                Assert.AreEqual(TestEnumValue, propertyBag.GetProperty(TestEnumOptionThree));
+
+                ValidateStringSet(StringSetValue, propertyBag.GetProperty(StringSetOption));
+
+                Assert.AreEqual(EmbeddedDoubleValue, propertyBag.GetProperty(PropertyBagOption).GetProperty(DoubleOption));
+                Assert.AreEqual(EmbeddedBooleanValue, propertyBag.GetProperty(PropertyBagOption).GetProperty(BooleanOption));
+                Assert.AreEqual(EmbeddedTestEnumValue, propertyBag.GetProperty(PropertyBagOption).GetProperty(TestEnumOptionThree));
+
+                ValidateStringSet(EmbeddedStringSetValue, propertyBag.GetProperty(PropertyBagOption).GetProperty(StringSetOption));
+
+                ValidatePropertyBag(PropertyBagValue, propertyBag.GetProperty(PropertyBagOption));
+                ValidatePropertyBag(TypedPropertyBagValue, propertyBag.GetProperty(TypedPropertyBagOption));
+            }
+        }
+
+        private static PerLanguageOption<bool> BooleanOption = 
+            new PerLanguageOption<bool>("TestFeature", "BooleanOption", false);
```

Comment:
`const string` to avoid repeating `"TestFeature"`.

`nameof` to avoid literal strings that recap property names.